agent/retry_join.go

```diff
@@ -52,23 +52,32 @@ func newDiscover() (*discover.Discover, error) {
 	)
 }
 
-func retryJoinAddrs(disco *discover.Discover, cluster string, retryJoin []string, logger *log.Logger) []string {
+func retryJoinAddrs(disco *discover.Discover, cluster string, retryJoin []string, logger hclog.Logger) []string {
 	addrs := []string{}
 	if disco == nil {
 		return addrs
 	}
 	for _, addr := range retryJoin {
 		switch {
 		case strings.Contains(addr, "provider="):
-			servers, err := disco.Addrs(addr, logger)
+			servers, err := disco.Addrs(addr, logger.StandardLogger(&hclog.StandardLoggerOptions{
+				InferLevels: true,
+			}))
 			if err != nil {
 				if logger != nil {
-					logger.Printf("[ERR] agent: Cannot discover %s %s: %s", cluster, addr, err)
+					logger.Error("Cannot discover address",
+						"address", addr,
+						"cluster", cluster,
```

Comment:
Isn't this `cluster` field already present due to the `With("cluster",...)` used when the derived logger is created?

Reply:
Yeah, so this will output `agent.cluster [LOGLEVEL] cannot discovery address cluster LAN/WAN`, the way that it is outputting will allow users to be able to search for the key cluster and parse by LAN/WAN if they want

Reply:
I meant that you were using `With()` to already give its sublogger the `cluster` param:
```
// Creates a sublogger that will always have the given key/value pairs
With(args ...interface{}) Logger
```

Reply:
*remove cluster k/v